src/main/java/com/amazon/ion/impl/bin/IonRawBinaryWriter_1_1.kt

```diff
@@ -0,0 +1,898 @@
+        var thisContainerTotalLength: Long = currentContainer.length
+
+        // If we have a data-model container with no child values, we can replace it with the prefixed, zero-length opcode.
+        if (currentContainer.numChildren == 0 && currentContainer.type >= TE_LIST) {
+            val zeroLengthOpcode = when (currentContainer.type) {
+                TE_LIST, PREFIXED_LIST, DELIMITED_LIST -> OpCode.LIST_LENGTH_0
+                TE_SEXP, PREFIXED_SEXP, DELIMITED_SEXP -> OpCode.SEXP_LENGTH_0
+                else -> OpCode.STRUCT_LENGTH_0
+            }
+            thisContainerTotalLength++ // For the opcode
+            buffer.truncate(currentContainer.position + 1)
+            buffer.writeUInt8At(currentContainer.position, zeroLengthOpcode.toLong())
+        } else when (currentContainer.type) {
+            TE_LIST, TE_SEXP -> {
+                // Add one byte to account for the op code
+                thisContainerTotalLength += currentContainer.metadataOffset + writeCurrentContainerLength(lengthPrefixPreallocation, currentContainer.numChildren.toLong())
+            }
+            PREFIXED_LIST,
+            PREFIXED_SEXP -> {
+                thisContainerTotalLength++
+                val contentLength = currentContainer.length
+                if (contentLength <= 0xF) {
+                    // Clean up any unused space that was pre-allocated.
+                    buffer.shiftBytesLeft(currentContainer.length.toInt(), lengthPrefixPreallocation)
+                    val zeroLengthOpCode = if (currentContainer.type == PREFIXED_LIST) OpCode.LIST_LENGTH_0 else OpCode.SEXP_LENGTH_0
+                    buffer.writeUInt8At(currentContainer.position, zeroLengthOpCode + contentLength)
+                } else {
+                    thisContainerTotalLength += writeCurrentContainerLength(lengthPrefixPreallocation)
+                }
+            }
+            DELIMITED_LIST,
+            DELIMITED_SEXP -> {
+                thisContainerTotalLength += 2 // For the start and end delimiters
+                buffer.writeByte(OpCode.DELIMITED_CONTAINER_END)
+            }
+            DELIMITED_STRUCT_SID_TO_FS,
+            DELIMITED_STRUCT_SID,
+            DELIMITED_STRUCT_FS -> {
+                // Need a sacrificial field name before the closing delimiter. We'll use $0.
+                // This works regardless of whether we're in FlexSym or SID mode.
+                buffer.writeByte(PrimitiveEncoder.FLEX_ZERO)
+                thisContainerTotalLength += 3 // For the start opcode, throwaway field name, and end marker
+                buffer.writeByte(OpCode.DELIMITED_CONTAINER_END)
+            }
+            PREFIXED_STRUCT_SID,
+            PREFIXED_STRUCT_SID_TO_FS -> {
+                // Add one byte to account for the op code
+                thisContainerTotalLength++
+                // Write closing delimiter if we're in a delimited container.
+                // Update length prefix if we're in a prefixed container.
```

Comment:
This is the prefixed code path, so these comments should be deleted or modified.